docs/job-writing-guide.md

```diff
@@ -0,0 +1,750 @@
+---
+sidebar_label: Job Writing Guide
+title: Job Writing Guide
+---
+
+<!-- TODO: work out where this goes in the nav bar. Top level I think. I probably want to merge my structural changes first. -->
+
+Workflow automation and data integration in Openfn is realised through the
+creation of Jobs.
+
+This guide will walk you through key concepts and best practices for job
+writing. It is suitable for new coders and experienced Javascript programmers.
+In fact, even if you're an experienced Javacript Developer, there a number of
+key patterns in the OpenFn ecosystem which it is important to learn.
+
+A Job is a bunch of Javascript code which performs a particular task, like
+fetching data from Salesforce or converting some JSON data to FHIR standard.
+
+Each job uses exactly one Adaptor (often called a "connector") to perform its
+task. The Adaptor provides a collection of helper functions which makes it easy
+to communicate with a data source.
+
+This guide applies equally to writing Jobs on the app (Lightning) or through the
+CLI.
+
+:::info Workflows Multiple jobs can be chained together in a Workflow. A common
+pattern is to use one job to fetch data from datasource A, one job to convert or
+transform that data to be compatible with datasource B, and a third job to
+upload the transformed data to datasource B.
+
+To learn more about workflow design and implementation, see
+[Build & Manage Workflows](/documentation/build/workflows) :::
```

Comment:
@josephjclark all links to other pages need to reference the specific `.md` file in the repo in order in order for the links to not break if we update how we manage doc versions in docusaurus. 

[See this Slack post](https://openfn.slack.com/archives/C06HA4WF35K/p1707811642208039) on the convention we're following for linking to other pages.

Reply:
I have pushed a fix for this, See [8aae57e](https://github.com/OpenFn/docs/pull/444/commits/8aae57e5022f4e9c86e1a68fd90f4e6bb8c372de) cc @josephjclark @aleksa-krolls